security/pkg/pki/util/keycertbundle.go

```diff
@@ -272,11 +272,12 @@ func TimeBeforeCertExpires(certBytes []byte, now time.Time) (time.Duration, erro
 	}
 
 	certExpiryTimestamp, err := extractCertExpiryTimestamp("cert", certBytes)
+
 	if err != nil {
 		return 0, fmt.Errorf("failed to extract cert expiration timestamp: %v", err)
 	}
 
-	certExpiry := time.Duration(certExpiryTimestamp-float64(now.Unix())) * time.Second
+	certExpiry := certExpiryTimestamp.Sub(now)
```

Comment:
I think this is more idiomatic because we're not converting back and forth with units. However, it does *add* nanosecond precision to the measurements.